test/dialing/resolver.spec.ts

```diff
@@ -100,6 +117,10 @@ describe('Dialing (resolvable addresses)', () => {
     // Resolver stub
     resolver.onCall(0).returns(Promise.resolve(getDnsRelayedAddrStub(remoteId)))
 
+    // create reservation on relay
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    await pEvent(remoteLibp2p.circuitService!, 'relay:reservation')
```

Comment:
Please do not disable linting.

```suggestion
    if (remoteLibp2p.circuitService == null) {
      throw new Error('remote libp2p had no circuit service')
    }

    // create reservation on relay
    await pEvent(remoteLibp2p.circuitService, 'relay:reservation')
```